### PR TITLE
Improve feedback from CI runs with all secrets

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

When starting the CI run with all secrets, add a comment to the original PR with a link to that run, instead of only adding a comment after such run would fail. Update the comment with the status once it finishes.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by ebyhr** [comment]
/test-with-secrets sha=1a56bc8816812b48e715d655addb2d930cf7eefa

File: .github/actions/update-check/action.yml [hunk 1]
```diff
@@ -83,21 +83,31 @@ runs:
             exclude_pull_requests: true
           });
 
-          const message = "The CI workflow run with tests that require additional secrets finished as " + process.env.conclusion + ": " + run.html_url
+          const started = "The CI workflow run with tests that require additional secrets has been started: " + run.html_url
+          const finished = "The CI workflow run with tests that require additional secrets finished as " + process.env.conclusion + ": " + run.html_url
           const comments = await github.paginate(github.rest.issues.listComments.endpoint.merge({
             ...context.repo,
             issue_number: process.env.number
           }))
```
Comment (hashhar) [inline .github/actions/update-check/action.yml:91]: unrelated but can we only list for "recent" comments instead of all comments on every invocation

File: .github/actions/update-check/action.yml [hunk 2]
```diff
@@ -83,21 +83,31 @@ runs:
...
-          const exists = comments.filter(comment => comment.body === message).length != 0
+          const comment = comments.find(comment => comment.body === started || comment.body === finished)
+
+          if (comment !== undefined) {
+            if (comment.body === finished) {
+              return;
+            }
```
Comment (hashhar) [inline .github/actions/update-check/action.yml:97]: Looks unreachable. Even if you comment twice for same SHA two different runs will be started if the automation hasn't posted a comment yet

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -534,6 +534,16 @@ jobs:
...
+      - name: Update PR check
+        uses: ./.github/actions/update-check
+        if: >-
+          always() &&
+          github.event_name == 'repository_dispatch' &&
+          github.event.client_payload.slash_command.args.named.sha != '' &&
+          github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha
+        with:
+          pull_request_number: ${{ github.event.client_payload.pull_request.number }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (hashhar) [inline .github/workflows/ci.yml:546]: this is unrelated right? Because we already invoke this from the `test` job.

**Review by hashhar — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (nineinchnick) [reply .github/workflows/ci.yml:546]: We need to invoke this action twice, to add the comment when the workflow starts (so as early as possible) and when it finishes (fails).

File: .github/actions/update-check/action.yml (hunk 2 above)
Comment (nineinchnick) [reply .github/actions/update-check/action.yml:97]: This action is called from the test jobs created from a matrix, so there can be more than one. This is done to signal the original PR about the failure as early as possible.

Comment (hashhar) [reply .github/actions/update-check/action.yml:97]: this one deserves comment IMO.

File: .github/actions/update-check/action.yml (hunk 1 above)
Comment (nineinchnick) [reply .github/actions/update-check/action.yml:91]: I'm not sure how that'd help and if it's possible at all. I don't see a method for that in https://octokit.github.io/rest.js/v19#issues-list-comments

File: .github/actions/update-check/action.yml (hunk 2 above)
Comment (nineinchnick) [reply .github/actions/update-check/action.yml:97]: In the workflow?

File: .github/actions/update-check/action.yml (hunk 1 above)
Comment (hashhar) [reply .github/actions/update-check/action.yml:91]: I was thinking about rate-limits but it appears we never hit those. There is a `since` parameter.

File: .github/actions/update-check/action.yml (hunk 2 above)
Comment (hashhar) [reply .github/actions/update-check/action.yml:97]: yes, in the body of this `if` block would make sense
